### ISSUE TITLE
Include an application but don't start it

### ISSUE THREAD

**Issue by vmsp** [ninenines/erlang.mk#776]
Mnesia requires me to run `mnesia:create_schema` before starting the application itself (`application:start(mnesia)`).

If I add `mnesia` to `LOCAL_DEPS` it won't work as it is automatically started. If I don't add it as a dependency it just tells me mnesia's functions are undefined.

So, how to I include Mnesia as a dependency but without starting it? Is there maybe a way to add dependencies to the application resource file's `included_applications` (http://erlang.org/doc/man/app.html)?

**Comment by essen** [comment]
Not really an answer to your question itself but there are other possible scenarios: creating the DB before starting the node if necessary, or stopping mnesia to create tables in your_application:start. The former is generally easier to deal with.

Otherwise you can always define a .app.src for your application and have mnesia listed as `{mnesia, load}` but you'll still have to start it in your application so you might want to just stop it there too when necessary.

**Comment by vmsp** [comment]
Yeah, the whole `create_schema` is kind of awkward and it feels like it should be a standalone step but I don't want to make deployment a two step process so the application needs to set itself up.

Stopping, setting up and then restarting is certainly a smart idea.

I was also wondering if there's any issue in doing `PROJECT_APP_EXTRA_KEYS = {included_applications, ['mnesia']}` or just adding `{mnesia, load}` to `relx.config`?

**Comment by essen** [comment]
Oh yeah that probably works too. I completely forgot. :-)

Also it doesn't have to be a two-step deployment, the node's start script can certainly be used to do more things before starting the node for real.

**Comment by essen** [comment]
I think we did that extra key thing so we don't have to worry about the rarely used keys so if that works for you, then it works as intended. :-)
